pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -586,8 +593,14 @@ private BrokerResponseNative handleRequest(long requestId, String query, JsonNod
     requestContext.setQueryProcessingTime(totalTimeMs);
     augmentStatistics(requestContext, brokerResponse);
 
+    // Extract source info from incoming request
+    Collection<String> remoteIps = null;
+    if (requesterIdentity != null) {
+      remoteIps = ((HttpRequesterIdentity) requesterIdentity).getHttpHeaders().get("X-Forwarded-For");
```

Comment:
Typically reverse proxies are used. So X-Forwarded-For will be populated. But do we want to populate this field with the clientIP if the request is not forwarded as well?